test/helpers/cilium.go

```diff
@@ -524,9 +524,20 @@ func (s *SSHMeta) ReportFailed(commands ...string) {
 	fmt.Fprint(wr, "StackTrace Ends\n")
 	s.ReportDump()
 	s.GatherLogs()
+	s.CheckLogsForDeadlock()
 	fmt.Fprint(wr, "===================== EXITING REPORT GENERATION =====================\n")
 }
 
+// CheckLogsForDeadlock checks if the logs for Cilium log messages that signify
+// that a deadlock has occurred.
+func (s *SSHMeta) CheckLogsForDeadlock() {
+	deadlockCheckCmd := fmt.Sprintf("sudo journalctl -au %s | grep -qi -B 5 -A 5 deadlock", DaemonName)
+	res := s.Exec(deadlockCheckCmd)
+	if res.WasSuccessful() {
+		log.Errorf("Deadlock during test run detected, check Cilium logs for context")
```

Comment:
Not sure if this is the best way TBH. Cilium only will be here in case of a test failed, so you'll never see the error at all

Reply:
Why would we never see the error? It will be logged to console.

Reply:
I'm thinking to implement something like `JustBeforeEach` in AfterEach, maybe it's the right place. Leave it like that and I'll check in the near future ;-)